### PR TITLE
Updated PHPUnit to 7.5.14

### PR DESCRIPTION
**Pull request by claudiosanches**
Updating PHPUnit to 7.5.14.

I placed a few notes for future reference in this PR.

Only failing unit tests for coding standards, since I haven't fix any in this PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ before_script:
     # Install WP Test suite, install PHPUnit globally:
     if [[ ! -z "$WP_VERSION" ]]; then
       bash tests/bin/install.sh woocommerce_test root '' localhost $WP_VERSION
-      composer global require "phpunit/phpunit=4.8.*|6.5.*"
+      composer global require "phpunit/phpunit=5.7.*|7.5.*"
```
Comment (claudiosanches) [inline .travis.yml]: PHPUnit 5.7 for PHP 5.6.

File: composer.json [hunk 2]
```diff
@@ -8,19 +8,19 @@
...
   "require-dev": {
     "myclabs/deep-copy": "1.7.0",
-    "phpunit/phpunit": "6.5.14",
+    "phpunit/phpunit": "7.5.14",
     "woocommerce/woocommerce-sniffs": "0.0.6"
   },
   "config": {
     "platform": {
-      "php": "7.0"
+      "php": "7.1"
```
Comment (claudiosanches) [inline composer.json]: I had to update it, since it will stop installing PHPUnit 7.5+

File: phpunit.xml [hunk 3]
```diff
@@ -7,7 +7,6 @@
 	convertNoticesToExceptions="true"
 	convertWarningsToExceptions="true"
 	verbose="true"
-	syntaxCheck="true"
```
Comment (claudiosanches) [inline phpunit.xml]: No more valid for PHPUnit 7.5+

File: tests/framework/class-wc-unit-test-case.php [hunk 4]
```diff
@@ -123,23 +123,4 @@ public function throwAnException( $message = null, $code = null ) {
...
 	}
-
-	/**
-	 * Backport assertNotFalse to PHPUnit 3.6.12 which only runs in PHP 5.2.
-	 *
-	 * @since  2.2
-	 * @param  mixed  $condition The statement to evaluate as not false.
-	 * @param  string $message   A message to display if the assertion fails.
-	 */
-	public static function assertNotFalse( $condition, $message = '' ) {
```
Comment (claudiosanches) [inline tests/framework/class-wc-unit-test-case.php]: We don't support PHP 5.2 anymore, so no need for it.

File: tests/includes/listener-loader.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
```
Comment (claudiosanches) [inline tests/includes/listener-loader.php]: Replaces our copy of `SpeedTrapListener` and includes WP Test lib SpeedTrapListener. Note that WP have a compatibility layer for it: http://develop.svn.wordpress.org/trunk/tests/phpunit/includes/listener-loader.php

File: tests/unit-tests/admin/reports/class-wc-tests-report-sales-by-date.php [hunk 6]
```diff
@@ -96,7 +96,7 @@ public function test_get_report_data() {
 			$data->coupons[0]->order_item_name,
 			'There should be a single coupon applied.'
 		);
-		$this->assertEquals( $data->coupons[0]->discount_amount, $data->total_coupons, 'Total discount amount.' );
+		$this->assertEquals( wc_format_decimal( $data->coupons[0]->discount_amount, '' ), $data->total_coupons, 'Total discount amount.' );
```
Comment (claudiosanches) [inline tests/unit-tests/admin/reports/class-wc-tests-report-sales-by-date.php]: PHPUnit 7.5 is more strict, before strings like `"99"` are equal to `"99.00"`, but now we need to make strict as well.

**Review by rodrigoprimo — APPROVED** [review]
LGTM

**Pull request merged**